### PR TITLE
195: MCQ vs SCQ fix

### PR DESCRIPTION
**Pull request by akshay-copods**

### CODE REVIEW

File: app/components/form/Dropdown.tsx [hunk 1]
```diff
@@ -25,6 +25,9 @@ function DropdownField({
   actionName?: string
 }) {
   const { t } = useTranslation()
+  const createQuestionDropdownOptions = data.filter(
+    (item) => item.value !== 'SINGLE_CHOICE'
```
Comment (richa-copods) [inline app/components/form/Dropdown.tsx:29]: Add ENUM here

File: app/components/sections/QuestionCard.tsx [hunk 2]
```diff
@@ -20,6 +20,12 @@ const QuestionCard = ({
   index: number
 }) => {
   const { t } = useTranslation()
+  const displayName =
+    question.questionType?.displayName === 'Multiple Choice'
```
Comment (richa-copods) [inline app/components/sections/QuestionCard.tsx:24]: Use ENUM for question type

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 3]
```diff
@@ -20,9 +20,13 @@ const AddQuestionInSection = () => {
   const { t } = useTranslation()
 
   const { sectionDetails, questionTypes } = useLoaderData()
-  const [selectedTypeOfQuestion, onQuestionTypeChange] = useState(
-    questionTypes[0].id
-  )
+  const [selectedTypeOfQuestion, onQuestionTypeChange] = useState(() => {
+    for (let qt of questionTypes) {
```
Comment (richa-copods) [inline app/components/sections/add-question/AddQuestionInSection.tsx:24]: Give proper name

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 4]
```diff
@@ -145,13 +149,19 @@ const AddQuestionInSection = () => {
       question,
       options: [],
       correctAnswer: [],
-      questionTypeId: selectedTypeOfQuestion,
+      questionTypeId:
+        answerCount === 1
+          ? questionTypes.find((item: any) => item.value === 'SINGLE_CHOICE')
```
Comment (richa-copods) [inline app/components/sections/add-question/AddQuestionInSection.tsx:154]: Give proper type. and use ENUM fir item.value

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/sections/add-question/AddQuestionInSection.tsx (hunk 4 above)
Comment (akshay-copods) [reply app/components/sections/add-question/AddQuestionInSection.tsx:154]: Done

File: app/components/sections/add-question/AddQuestionInSection.tsx (hunk 3 above)
Comment (akshay-copods) [reply app/components/sections/add-question/AddQuestionInSection.tsx:24]: Done

File: app/components/sections/QuestionCard.tsx (hunk 2 above)
Comment (akshay-copods) [reply app/components/sections/QuestionCard.tsx:24]: Done

File: app/components/form/Dropdown.tsx (hunk 1 above)
Comment (akshay-copods) [reply app/components/form/Dropdown.tsx:29]: Done

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 5]
```diff
@@ -20,9 +20,13 @@ const AddQuestionInSection = () => {
   const { t } = useTranslation()
 
   const { sectionDetails, questionTypes } = useLoaderData()
-  const [selectedTypeOfQuestion, onQuestionTypeChange] = useState(
-    questionTypes[0].id
-  )
+  const [selectedTypeOfQuestion, onQuestionTypeChange] = useState(() => {
+    for (let questionType of questionTypes) {
+      if (questionType.value == 'MULTIPLE_CHOICE') {
```
Comment (richa-copods) [inline app/components/sections/add-question/AddQuestionInSection.tsx:25]: Use ENUM for "MULTIPLE_CHOICE"

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 6]
```diff
@@ -77,7 +82,6 @@ const AddQuestionInSection = () => {
     return quesValue
   }
 
-  const submit = useSubmit()
   const saveQuestion = (addMoreQuestion: boolean) => {
     if (!question.length) {
       toast.error('Enter the Question', { toastId: 'questionRequired' })
```
Comment (richa-copods) [inline app/components/sections/add-question/AddQuestionInSection.tsx:87]: Create constant for this

File: app/components/sections/QuestionCard.tsx [hunk 7]
```diff
@@ -20,6 +20,12 @@ const QuestionCard = ({
   index: number
 }) => {
   const { t } = useTranslation()
+  const displayName =
+    question.questionType?.displayName === QuestionTypes.multipleChoice
+      ? { name: 'MSQ', full: 'Multiple Select Question' }
```
Comment (richa-copods) [inline app/components/sections/QuestionCard.tsx:25]: Use constants here

File: app/components/form/Dropdown.tsx [hunk 8]
```diff
@@ -96,7 +99,7 @@ function DropdownField({
                     </div>
                   </Listbox.Option>
                 )}
-                {data.map((el) => (
+                {createQuestionDropdownOptions.map((el) => (
```
Comment (richa-copods) [inline app/components/form/Dropdown.tsx:102]: give proper name instead of el

File: app/components/sections/QuestionCard.tsx [hunk 9]
```diff
@@ -54,9 +60,10 @@ const QuestionCard = ({
           <div>
             <span
               id="question-type"
+              title={displayName?.full}
               className="flex flex-1 items-center rounded-52 border border-gray-700 px-3 text-sm text-gray-700"
             >
-              {question.questionType?.displayName}
+              {displayName?.name}
```
Comment (richa-copods) [inline app/components/sections/QuestionCard.tsx:66]: Use translation

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/sections/add-question/AddQuestionInSection.tsx (hunk 5 above)
Comment (akshay-copods) [reply app/components/sections/add-question/AddQuestionInSection.tsx:25]: Done

File: app/components/sections/add-question/AddQuestionInSection.tsx (hunk 6 above)
Comment (akshay-copods) [reply app/components/sections/add-question/AddQuestionInSection.tsx:87]: Done

File: app/components/sections/QuestionCard.tsx (hunk 7 above)
Comment (akshay-copods) [reply app/components/sections/QuestionCard.tsx:25]: Done

File: app/components/form/Dropdown.tsx (hunk 8 above)
Comment (akshay-copods) [reply app/components/form/Dropdown.tsx:102]: done

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 10]
```diff
@@ -20,9 +20,13 @@ const AddQuestionInSection = () => {
   const { t } = useTranslation()
 
   const { sectionDetails, questionTypes } = useLoaderData()
-  const [selectedTypeOfQuestion, onQuestionTypeChange] = useState(
-    questionTypes[0].id
-  )
+  const [selectedTypeOfQuestion, onQuestionTypeChange] = useState(() => {
+    for (let questionType of questionTypes) {
+      if (questionType.value == QuestionTypes.multipleChoice) {
```
Comment (richa-copods) [inline app/components/sections/add-question/AddQuestionInSection.tsx]: Use strict check ===

**Review by richa-copods — CHANGES_REQUESTED** [review]

**Review by richa-copods — APPROVED** [review]

**Pull request merged**